nsxt/data_source_nsxt_failure_domain.go

```diff
@@ -0,0 +1,99 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+func dataSourceNsxtFailureDomain() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNsxtFailureDomainRead,
+
+		Schema: map[string]*schema.Schema{
+			"id":           getDataSourceIDSchema(),
+			"display_name": getDataSourceDisplayNameSchema(),
+			"description":  getDataSourceDescriptionSchema(),
+			"preferred_active_edge_services": {
+				Type:        schema.TypeString,
```

Comment:
As the definition of `preferred_active_edge_services` [here](https://vdc-repo.vmware.com/vmwb-repository/dcr-public/612caf56-5bb1-4a0b-8d2b-12ccd8981904/37be75ba-aa56-42f4-99c3-f6c989ddbe57/api_includes/types_FailureDomain.html ). `true`, `false` and `unset`, all three are valid cases. nil pointer in the SDK represents the `unset` state. So I use `TypeString` instead of `TypeBool` here.

Reply:
Understood. But shouldn't this be computed instead of optional?

Reply:
I think it's more optional than computed. Basically, it takes what users provide and goes with no preference, nil on API side, if users don't set this field.

Reply:
But this is a data source - so each field is either a search argument (and then required or optional) or an output (and then computed - as it's only set by TF). Maybe I'm missing something (sorry I'm new to TF...).

Reply:
I don't think this field is needed in the data source, but if present, it should be indeed Computed

Reply:
My bad. I didn't notice we were discussing the data source instead of the resource. 
Yes, makes sense. Removed this field.